probatus/feature_elimination/feature_elimination.py

```diff
@@ -695,3 +695,157 @@ def plot(self, show=True, **figure_kwargs):
+                models, such as xgboost and LightGBM.
+
+            eval_metric (str, optional):
+                Metric for early stopping.
+        """  # noqa
+        super(EarlyStoppingShapRFECV, self).__init__(clf, **kwargs)
+        if early_stopping_rounds:
+            print(
+                "WARNING: early_stopping_rounds will be passed when estimating feature importance with SHAP,"
+                + "but not when optimizing hyperparameters."
+            )
+        self.early_stopping_rounds = early_stopping_rounds
+        self.eval_metric = eval_metric
+
+    def _get_feature_shap_values_per_fold(self, X, y, clf, train_index, val_index, scorer, verbose=0, **shap_kwargs):
+        """
+        This function calculates the shap values on validation set, and Train and Val score.
+
+        Args:
+            X (pd.DataFrame):
+                Dataset used in CV.
+
+            y (pd.Series):
+                Binary labels for X.
+
+            clf (binary classifier):
+                Model to be fitted on the train folds.
+
+            train_index (np.array):
+                Positions of train folds samples.
+
+            val_index (np.array):
+                Positions of validation fold samples.
+
+            scorer (string, callable or None):
+                A string (see sklearn [model scoring](https://scikit-learn.org/stable/modules/model_evaluation.html)) or
+                a scorer callable object, function with the signature `scorer(estimator, X, y)`.
+
+            verbose (int, optional):
+                Controls verbosity of the output:
+
+                - 0 - neither prints nor warnings are shown
+                - 1 - 50 - only most important warnings regarding data properties are shown (excluding SHAP warnings)
+                - 51 - 100 - shows most important warnings, prints of the feature removal process
+                - above 100 - presents all prints and all warnings (including SHAP warnings).
+
+            **shap_kwargs:
+                keyword arguments passed to
+                [shap.Explainer](https://shap.readthedocs.io/en/latest/generated/shap.Explainer.html#shap.Explainer).
+                It also enables `approximate` and `check_additivity` parameters, passed while calculating SHAP values.
+                The `approximate=True` causes less accurate, but faster SHAP values calculation, while
+                `check_additivity=False` disables the additivity check inside SHAP.
+        Returns:
+            (np.array, float, float):
+                Tuple with the results: Shap Values on validation fold, train score, validation score.
+        """
+        X_train, X_val = X.iloc[train_index, :], X.iloc[val_index, :]
+        y_train, y_val = y.iloc[train_index], y.iloc[val_index]
+
+        # Fit model with train folds
+        if self.early_stopping_rounds:
+            clf = clf.fit(
```

Comment:
I have another idea: 
How about we package lines between 833 and 843 into a separate function `_fit_model`. 
This way we don't need to copy paste the code of `_get_feature_shap_values_per_fold` and only make sure we pass kwargs to the `_fit_model` correctly. What do you think?

The main reason for doing this is that we also need to pass early stopping to `fit` of ShapRFECV in lines 502 and 506.